executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -217,62 +218,42 @@ impl<'a, T: FieldElement> FixedLookup<'a, T> {
         }
 
         // Split the left-hand-side into known input values and unknown output expressions.
-        let mut input_values = vec![];
-        let mut known_inputs: BitVec = Default::default();
-        let mut output_expressions = vec![];
-
-        for l in left {
-            if let Some(value) = l.constant_value() {
-                input_values.push(value);
-                known_inputs.push(true);
-            } else {
-                output_expressions.push(l);
-                known_inputs.push(false);
-            }
-        }
-
-        let application = Application {
-            identity_id,
-            inputs: known_inputs,
-        };
-
-        let index = self
-            .indices
-            .entry(application)
-            .or_insert_with_key(|application| {
-                create_index(self.fixed_data, application, &self.connections)
-            });
-        let index_value = index.get(&input_values).ok_or_else(|| {
-            let input_assignment = left
-                .iter()
-                .zip(right)
-                .filter_map(|(l, r)| l.constant_value().map(|v| (r.name.clone(), v)))
-                .collect();
-            EvalError::FixedLookupFailed(input_assignment)
-        })?;
+        let mut data = vec![T::zero(); left.len()];
+        let values = left
+            .iter()
+            .zip(&mut data)
+            .map(|(l, d)| {
+                if let Some(value) = l.constant_value() {
+                    *d = value;
+                    LookupCell::Input(d)
+                } else {
+                    LookupCell::Output(d)
+                }
+            })
+            .collect::<Vec<_>>();
 
-        let Some((row, output)) = index_value.get() else {
+        if !self.process_lookup_direct(mutable_state, identity_id, values)? {
             // multiple matches, we stop and learnt nothing
             return Ok(EvalValue::incomplete(
                 IncompleteCause::MultipleLookupMatches,
             ));
         };
 
-        self.multiplicity_counter.increment_at_row(identity_id, row);
-
         let mut result = EvalValue::complete(vec![]);
```

Comment:
Would it help to provide a capacity of `left.len()` here? I think `evaluated.solve()` (as opposed to `solve_with_range_constraints`) would return at most one assignment, right?